src/Data/List/Infinite.hs

```diff
@@ -605,6 +606,14 @@ scanl1 f (x :< xs) = scanl f x xs
 mapAccumL :: (acc -> x -> (acc, y)) -> acc -> Infinite x -> Infinite y
 mapAccumL f = flip (foldr (\x acc s -> let (s', y) = f s x in y :< acc s'))
 
+-- | Same as 'mapAccumL', but strict in accumulator.
+mapAccumL' :: (acc -> x -> (acc, y)) -> acc -> Infinite x -> Infinite y
+mapAccumL' f = go
```

Comment:
Could it be expressed via `foldr` or other high-level combinator? Could there be rewrite rules for fusion similar to `mapAccumLFB`?

Reply:
I'll take a look.

Reply:
@Bodigrim 

I wrote `mapAccumL'` following the definition for `mapAccumL`, and simply adding bang patterns on the accumulator. This works (not getting stack overflows, at least for now), but [this benchmark](https://github.com/Daniel-Diaz/infinite-list/blob/mapAccumLBench/bench/MapAccumL.hs) shows the `foldr` version taking ~15% longer when I run it. I don't know whether I'm doing something wrong.

Reply:
The benchmark gives the same numbers on my machine, at least with GHC 9.10.2:
```
benchmarking mapAccumL/mapAccumL1
time                 30.84 μs   (30.80 μs .. 30.91 μs)
                     1.000 R²   (1.000 R² .. 1.000 R²)
mean                 30.89 μs   (30.84 μs .. 30.96 μs)
std dev              181.4 ns   (141.0 ns .. 232.0 ns)

benchmarking mapAccumL/mapAccumL2
time                 30.95 μs   (30.88 μs .. 31.04 μs)
                     1.000 R²   (1.000 R² .. 1.000 R²)
mean                 31.04 μs   (30.90 μs .. 31.38 μs)
std dev              646.1 ns   (180.1 ns .. 1.155 μs)
variance introduced by outliers: 18% (moderately inflated)
```
So let's proceed with the `foldr` version.

Could you please add tests, both to `Properties.hs` and `Fusion.hs`?

Reply:
Thanks. That's good to know. I was using ghc-9.8.4. I'll see to add those tests later today.

Reply:
@Bodigrim 

Tests added in https://github.com/Bodigrim/infinite-list/pull/22/commits/90e40ab91576fd0919ca2466efd733c7cd965515.